### ISSUE TITLE
Correcting for multiple batches

### ISSUE THREAD

**Issue by FrancesWong** [LTLA/batchelor#10]
Hi, 

I was hoping you could give some advice for how to correct for multiple batches. I have batches from experimental procedures (samples processed in two batches) and due to a sequencing error, the sequencing was also done in two batches. The breakdown is as follows (Y axis - experimental batches, X axis - sequencing batches)

    Batch1 Batch2
  1     19     23
  2     15      5

Currently, I've read in the two sequencing runs separately, correct on them (correction worked beautifully!), extracted the data and created two new separate objects based on the experimental batches. The second batch did not correct very much. I'm used to doing my scSEQ analysis on Seurat so after batch correcting with batchelor, I extracted the corrected counts to make a Seurat object so after the first correction, Seurat is still showing an influence from experimental batch but when I look at the pre-correction PCA separated on experimental batch with the scater/batchelor workflow, it doesn't detect a difference. 

I can provide some code and figures if it would help but don't want to bore you if the approach is completely wrong. 

Your advice is greatly appreciated! 

Thanks, 
-Frances

**Comment by LTLA** [comment]
- I have no idea what those numbers are. 
- _batchelor_ has several correction methods, you'll have to be more specific.
- I don't know what you mean by "correct on them", you'll have to be more specific.

**Comment by FrancesWong** [comment]
Hi, 

Sorry about the confusion and my lack of clarity, please let me explain in more detail. 

There are two groups of batches I'm trying to correct (or rescale) for. The experiment was conducted in two sequencing runs. Batch1 had 34 samples where as Batch2 had 28 samples. There is another layer of experimental batch because the extraction was done in two steps, so of Batch1, 19 were in ExtractionA and 15 were part of ExtractionB. Similarly from Batch2, 23 were from ExtractionA and 5 were from ExtractionB. 

My strategy so far was to rescale on sequencing batch and then rescale on extraction batch. 

Essentially, I have the two sequencing batches as SingleCellExperiment objects sceO and sceR. I ran
rescaleBatches(sceO, sceR), then I extract the matrix and cut it into two objects based on the Extraction batches and run rescaleBatches(extA, extB) to rescale the second factor. 

Is this the right approach? 

Thanks! (Sorry I'll be away from the lab tomorrow so it might be Thus until I can get you back a reply!) 
-Frances

**Comment by LTLA** [comment]
There are several points here. First, your immediate technical problems. 

- If I was to do it this way, I would be inclined to define four "batches", one batch per combination of extraction and sequencing. Then I would perform correction on those four batches.
- There is no need to literally cut up the SCE, as `rescaleBatches` has a `batch=` argument that you can use to pass in a single SCE and get a single SCE out that contains the corrected values.

Now, the second level of problems:

- If the only experimental difference between sequencing batches is the sequencing itself, that probably doesn't introduce strong batch effects, so correction is usually trivial. I don't know what you mean by extraction, but if you're talking about dissociation from tissue, then I could imagine that would introduce batch effects. Perhaps it's not surprising that the correction is more difficult for that difference.
- As `?rescaleBatches` states, it assumes that each batch contains the same population composition. This is generally pretty a pretty strong (i.e., easily violated) assumption. For example, if a certain cell type is less efficiently extracted in the second-extraction batches, then the assumption no longer holds. More generally, this assumption usually doesn't hold for merges of biological replicates because cell composition naturally varies from animal to animal (see [here](https://osca.bioconductor.org/integrating-datasets.html#linear-regression) for a discussion).

And onto the final problem:

- You are assuming that you don't have batch effects between samples within each batch. I don't know what these samples are - are they individual 10X runs? - but that seems like something that you ought to check at some point.

**Comment by FrancesWong** [comment]
Hi, 

Thank you for your detailed reply. 

Sorry for the delayed reply, I spent the day trying to execute it. Separating it into 4 batches solved the problem nicely! Thanks! 

When I plotted a PCA before rescalling for sequencing run, the samples separated on that clearly (they were done weeks apart so many parameters could have changed between). The extraction is the RNA extraction from the cells. The 4 group set up seems to have corrected it well. 

The samples were arrayed into 96well plates for smart-seq. I will keep in mind to check for these batches and also for the population composition.

Thanks again, I really appreciate your help! 

Best, 
-Frances
